app/src/main/java/org/mozilla/focus/utils/UrlUtils.java

```diff
@@ -62,6 +64,74 @@ public static String createSearchUrl(Context context, String searchTerm) {
         return searchEngine.buildSearchUrl(searchTerm);
     }
 
+    /**
+     * @return The search terms for the first search performed with the default search engine or the current URL.
+     */
+    public static String getSearchTermsOrUrl(Context context, String url) {
```

Comment:
Can you file a follow-up bug that we write some unit tests for those new helper methods? I'll merge the patch before that so that we get some testing time from QA already.